src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison/PrisonApiClient.kt

```diff
@@ -27,6 +28,41 @@ class PrisonApiClient(@Qualifier("oauthPrisonClient") val prisonerApiWebClient:
       .onErrorResume { webClientErrorHandler(it) }
   }
 
+  fun hdcStatuses(bookingIds: List<Long>): List<PrisonerHdcStatus> {
+    return prisonerApiWebClient
+      .post()
+      .uri("/offender-sentences/home-detention-curfews/latest")
+      .accept(MediaType.APPLICATION_JSON)
+      .bodyValue(bookingIds)
+      .retrieve()
+      .bodyToMono(typeReference<List<PrisonerHdcStatus>>())
+      .onErrorResume { webClientErrorHandler(it) }
+      .block() ?: emptyList<PrisonerHdcStatus>()
+  }
+
+  fun offenceHistories(bookingIds: List<Long>): List<PrisonerOffenceHistory> {
+    return prisonerApiWebClient
+      .post()
+      .uri("/bookings/offence-history")
+      .bodyValue(bookingIds)
+      .accept(MediaType.APPLICATION_JSON)
+      .retrieve()
+      .bodyToMono(typeReference<List<PrisonerOffenceHistory>>())
+      .onErrorResume { webClientErrorHandler(it) }
+      .block() ?: emptyList<PrisonerOffenceHistory>()
+  }
+
+  fun getIS91AndExtraditionBookingIds(bookingIds: List<Long>): List<Long> {
```

Comment:
this is a bit of business code and probably belongs in the service layer with some tests.